### PR TITLE
adding images for skill assessment

### PR DESCRIPTION
**Pull request by mariajmolina**
Adding images for the input, output, and label for skill assessment. Also creates a trial number folder within directory to place images in.

### CODE REVIEW

File: applications/train.py [hunk 1]
```diff
@@ -68,8 +69,12 @@ def train_one_epoch(model, dataloader, optimizer, criterion, nc, clip=1.0):
...
 
     Args:
-        model (torch): pytorch neural network.
-        dataloader (torch): pytorch dataloader.
+        model (torch): pytorch neural network
+        dataloader (torch): pytorch dataloader
+        optimizer: training optimizer
+        criterion: loss function
+        nc: number of channels
+        clip: clipping of gradients, defaults to 1.0
```
Comment (jhayron-perez) [inline applications/train.py:77]: Nice to have this more detailed description!

File: applications/train.py [hunk 2]
```diff
@@ -160,6 +174,41 @@ def validate(model, dataloader, criterion, metrics, second_metrics, nc):
...
+        plt.close()
+
+        # label
+        c_lbl = img_label.cpu().detach().numpy()[:,0,:,:][samp_]
+        im = plt.imshow(c_lbl)
+        plt.colorbar(im)
+        plt.savefig(
+            f"{save_loc}/trial{str(trial_num)}/{data_split}_lbl_{str(epoch)}_{str(trial_num)}.png", 
+            bbox_inches='tight')
+        plt.close()
```
Comment (jhayron-perez) [inline applications/train.py]: Looks great! I wonder if it would be better to make this section optional, could depend on one argument. Just in case we'd like to run validations without plotting anything.

File: applications/train.py [hunk 3]
```diff
@@ -359,16 +408,21 @@ def trainer(conf, trial=False, verbose=True):
...
+    os.makedirs(save_loc+"/trial"+str(int(trial.number)), exist_ok=True)
+    
     for epoch in list(range(epochs)):
         
         t_loss, t_corr, t_true = train_one_epoch(
             model, train_loader, optimizer, train_loss, nc
         )
         v_loss, v_corr, v_true, metrics, cesm_metrics = validate(
-            model, valid_loader, valid_loss, validation_metrics, validation_metrics_cesm, nc
+            model, valid_loader, valid_loss, validation_metrics, validation_metrics_cesm,
```
Comment (jhayron-perez) [inline applications/train.py:436]: Similar to the previous comment. Should we include a way of making the plotting just for some epochs/trials? Could it be possible that this generates "too many" images?

File: applications/train.py (hunk 2 above)
Comment (mariajmolina) [reply applications/train.py]: good point -- I am also getting annoyed with how many images are getting saved each epoch for EACH trial lol... I think every fifth epoch or something may be better

File: applications/train.py (hunk 3 above)
Comment (mariajmolina) [reply applications/train.py:436]: Agreed! See comment above -- I am running a study now and it is A LOT of images...

**Review by jhayron-perez — COMMENTED** [review]
This is so useful, Maria! I'm excited to be able to visualize the results. I only wonder if we should make these changes "optional," just in case we'd like to run the plotting just for some epochs or trials. :)

**Comment by mariajmolina** [comment]
> 

Added option for image generation from model yaml file, with option for user defined spacing.
Also caught/fixed a bug in the dataset loader where precip unit conversion was also needed in the norm functions.

**Review by jhayron-perez — APPROVED** [review]
Looks great!!
